script.js

```diff
@@ -148,13 +148,40 @@ modalCancel.addEventListener('click', () => {
   header.classList.toggle('blur');
 });
 
+const fullName = document.getElementById('name');
+const email = document.getElementById('email');
+const message = document.getElementById('message');
+
+let formData = {
+  fullName: '',
+  email: '',
+  message: '',
+};
+
 form.addEventListener('submit', (event) => {
-  const email = form.elements.email.value;
   const validation = document.querySelector('.validation');
-  if (email !== email.toLowerCase()) {
+  if (email.value !== email.value.toLowerCase()) {
     event.preventDefault();
     validation.innerText = 'Please type email address in lower case!';
   } else {
     form.submit();
   }
+});
+
+function saveForm() {
+  formData.fullName = fullName.value;
+  formData.email = email.value;
+  formData.message = message.value;
+  localStorage.setItem('formLocalStorage', JSON.stringify(formData));
+}
```

Comment:
- [ ] kindly consider making sure that as I type a single letter in any input it's immediately saved in the local storage so that After filling out the form, and not submitting it, the entered values should be saved in the browser and when visiting the page the form inputs should be contained or auto-populated with, the values entered in the previous visit to the page. because currently the data are not being saved at all

Reply:
Hi @divinecharlotte,

Thank you for your feedback. 

My inputs are saving every single letter as Im triggering with input in the fields.
those lines are slightly below where you marked.